src/osip-integration-test/src/test/java/edu/kit/pse/osip/monitoring/SimulationControllerTest.java

```diff
@@ -0,0 +1,77 @@
+package edu.kit.pse.osip.monitoring;
+
+import edu.kit.pse.osip.core.OSIPConstants;
+import edu.kit.pse.osip.core.io.files.ServerSettingsWrapper;
+import edu.kit.pse.osip.core.io.networking.RemoteMachine;
+import edu.kit.pse.osip.core.model.base.TankSelector;
+import edu.kit.pse.osip.core.opcua.client.UAClientException;
+import edu.kit.pse.osip.monitoring.controller.MixTankClient;
+import edu.kit.pse.osip.monitoring.controller.TankClient;
+import edu.kit.pse.osip.simulation.controller.SimulationController;
+import java.io.File;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import javafx.stage.Stage;
+import org.junit.AfterClass;
+import org.junit.Test;
+import org.testfx.framework.junit.ApplicationTest;
+
+import static org.junit.Assert.assertTrue;
+
+/**
+ * A class to test the SimulationController class.
+ *
+ * @author David Kahles
+ * @version 1.0
+ */
+public class SimulationControllerTest extends ApplicationTest {
+    private static CompletableFuture<Boolean> closed = new CompletableFuture<>();
+    private static SimulationController simCont;
+
+    @Override
+    public void start(Stage primaryStage) {
+        simCont = new SimulationController();
+        simCont.start(primaryStage);
+    }
+
+    /**
+     * Start the SimulationController and check whether we can connect OPC UA Clients to it. Don't test
+     * the GUI any further. Just make sure it doesn't throw an exception.
+     * @throws InterruptedException If the future has an error.
+     * @throws UAClientException If connecting to the OPC UA servers fails.
+     */
+    @Test(timeout = 30000)
+    public void testLaunch() throws InterruptedException, UAClientException {
+        Thread.sleep(5000);
+
+        File settingsLocation = new File(System.getProperty("user.home") + File.separator + ".osip");
+        settingsLocation.mkdirs();
+        ServerSettingsWrapper settings = new ServerSettingsWrapper(new File(settingsLocation, "simulation.conf"));
+
+        MixTankClient mixClient = new MixTankClient(new RemoteMachine("127.0.0.1",
+            settings.getServerPort(TankSelector.MIX, OSIPConstants.DEFAULT_PORT_MIX)));
+        TankClient client = new TankClient(new RemoteMachine("127.0.0.1",
+            settings.getServerPort(TankSelector.valuesWithoutMix()[0], OSIPConstants.DEFAULT_PORT_MIX + 1)));
+
+        mixClient.connectClient();
+        client.connectClient();
+        mixClient.disconnectClient();
+        client.disconnectClient();
+    }
+
+    /**
+     * Check whether the application got stopped
+     * @throws InterruptedException If the CompletableFuture has an error.
+     * @throws ExecutionException If the CompletableFuture has an error.
+     */
+    @AfterClass
+    public static void testClosed() throws InterruptedException, ExecutionException {
+        assertTrue(closed.get());
```

Comment:
Irgendwie finde ich das nicht so super. AfterClass hat soweit ich weiß kein Timeout, also wird das hier endlos hängen, wenn `stop()` fehlschlägt. Was spricht dagegen, stop() direkt in testLaunch aufzurufen?